src/object/cli_shard.c

```diff
@@ -755,7 +755,13 @@ dc_obj_shard_list(struct dc_obj_shard *obj_shard, enum obj_rpc_opc opc,
 		oei->oei_akey = *obj_args->akey;
 	oei->oei_oid		= obj_shard->do_id;
 	oei->oei_map_ver	= args->la_auxi.map_ver;
-	oei->oei_epoch		= args->la_auxi.epoch;
+	if (obj_args->eprs != NULL && opc == DAOS_OBJ_RPC_ENUMERATE) {
+		oei->oei_epr = *obj_args->eprs;
+	} else {
+		oei->oei_epr.epr_lo = args->la_auxi.epoch;
+		oei->oei_epr.epr_hi = args->la_auxi.epoch;
```

Comment:
I think all kinds of enumeration should support epoch range, if the interface specify only one epoch, the range should be [0, epoch), right?

Reply:
indeed.